app/models/maestrano/connector/rails/concerns/sub_entity_base.rb

```diff
@@ -34,72 +34,43 @@ def names_hash
       end
     end
 
-    def create_idmap_from_external_entity(entity, connec_entity_name, organization)
-      if external?
-        h = names_hash.merge({
-          external_id: id_from_external_entity_hash(entity),
-          name: object_name_from_external_entity_hash(entity),
-          connec_entity: connec_entity_name.downcase,
-          organization_id: organization.id
-        })
-        Maestrano::Connector::Rails::IdMap.create(h)
-      else
-        raise 'Forbidden call: cannot call create_idmap_from_external_entity for a connec entity'
-      end
-    end
-
-    def create_idmap_from_connec_entity(entity, external_entity_name, organization)
-      if external?
-        raise 'Forbidden call: cannot call create_idmap_from_connec_entity for an external entity'
-      else
-        h = names_hash.merge({
-          connec_id: entity['id'],
-          name: object_name_from_connec_entity_hash(entity),
-          external_entity: external_entity_name.downcase,
-          organization_id: organization.id
-        })
-        Maestrano::Connector::Rails::IdMap.create(h)
-      end
-    end
-
     # { 'External Entity'  => LalaMapper, 'Other external entity' => LiliMapper }
     # or { 'Connec Entity'  => LalaMapper, 'Other connec entity' => LiliMapper }
     def mapper_classes
       {}
     end
 
     # {
-    #   'External Entity' => [{reference_class: Entities::.., connec_field: '', external_field: ''}],
-    #   'Other external entity' => [{reference_class: Entities::.., connec_field: '', external_field: ''}]
+    #   'External Entity' => ['organization_id'],
+    #   'Other external entity' => ['an array of the connec reference fields']
     # }
     def references
       {}
     end
   end
 
 
-  def map_to(name, entity, organization)
+  def map_to(name, entity)
     mapper = self.class.mapper_classes[name]
     raise "Impossible mapping from #{self.class.entity_name} to #{name}" unless mapper
 
-    ref_hash = {}
-    if self.class.references[name]
-      self.class.references[name].each do |ref|
-        field = self.class.external? ? ref[:connec_field] : ref[:external_field]
-        ref_hash.merge! field.split('/').reverse.inject(Maestrano::Connector::Rails::Entity.id_from_ref(entity, ref, self.class.external?, organization)) { |a, n| { n.to_sym => a } }
-      end
-    end
-
     if self.class.external?
-      mapped_entity = mapper.denormalize(entity)
+      mapped_entity = mapper.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity))
+      folded_entity = Maestrano::Connector::Rails::ConnecHelper.fold_references(mapped_entity, self.class.references[name] || [], @organization)
+      folded_entity.merge!(opts: (mapped_entity[:opts] || {}).merge(matching_fields: self.class.connec_matching_fields)) if self.class.connec_matching_fields
```

Comment:
Here, if self.class.connec_matching_fields is nil, folded_entity won't be merge to opts: (mapped_entity[:opts] || {}, 

Is it wanted ?